ledger/src/ui/src/pin.h

```diff
@@ -29,6 +29,67 @@
 
 #define PIN_LENGTH 8
 
-bool is_pin_valid(unsigned char *pin);
+// Pin context
+typedef struct {
+    unsigned char* pin_buffer;
+} pin_t;
+
+// Helper macros for pin context manipulation
+#define PIN_CTX_PAYLOAD(pin_ctx) ((unsigned char*)((pin_ctx)->pin_buffer + 1))
```

Comment:
Two things that are not super relevant, but might aid in readability:

1. I feel like we need a high level explanation of the two pin versions we use, and where each is used (i.e., with prepended length and null-terminated).
2. The names `PIN_CTX_PAYLOAD` and `PIN_CTX_PAYLOAD_LEN` might be a bit over complicated. Maybe something simpler and more to the point? Like `GET_PIN` and `GET_PIN_LENGTH`? Or something along those lines?

What do you think?